app/javascript/components/rooms/room/RoomJoin.jsx

```diff
@@ -69,6 +97,28 @@ export default function RoomJoin() {
     return <Navigate replace to="/" />;
   }
 
+  const hasAccessCode = publicRoom.data?.viewer_access_code || publicRoom.data?.moderator_access_code;
+
+  if (!publicRoom.data?.viewer_access_code && publicRoom.data?.moderator_access_code) {
+    fields.accessCode.label = 'Moderator Access Code (Optional)';
+  } else {
+    fields.accessCode.label = 'Access Code';
+  }
```

Comment:
What if it has both? Is that handled here?

Reply:
Yup, it's the same condition that we had before this minor refactoring.
If there's only a moderator access code in place the label will change otherwise it will be **Access Code**, when there no codes in place the input will be unmounted.